airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java

```diff
@@ -314,6 +314,24 @@ public List<CheckedConsumer<JdbcDatabase, Exception>> getCheckOperations(final J
       checkOperations.add(database -> {
         PostgresUtils.checkFirstRecordWaitTime(config);
       });
+
+      checkOperations.add(database -> {
+        final String userName = config.get("username").toString();
+
+        final List<JsonNode> userPrivileges = database.queryJsons(connection -> {
+          final PreparedStatement ps = connection.prepareStatement("SELECT userepl FROM pg_user WHERE usename = ?");
```

Comment:
can we query the `pg_user` table with the permissions that we ask for?

Reply:
Yes we can.
In order to test this I just created a new user without any privilege (neither REPLICATE or table access), and the user is able to query it.
I think that by default, all users can query that table.